### DISCUSSION TITLE
Add default folder for user profile images

### DISCUSSION THREAD

**Discussion by lucameusburger** [directus/directus#12019] [Feature Requests]
Hey there!

First of all - i love directus and the team behind it. Keep it up!

I manage all files with folders and every file-field has a default folder where stuff gets stored. So the first layer in the filemanager is always empty.

Exept when a user uploads a profile picture.

Would be nice to define the default folder for those.

I think it would be a breaking change to make the field 'avatar' in 'directus-users' editable. Maybe but it in the project-settings.

![image](https://user-images.githubusercontent.com/25350597/157100628-33dc8e9a-631d-467a-9661-8e0e9a0b9dea.png)

**Comment by azrikahar** [comment]
It'd be nice indeed! In the meantime, since in your specific use case you are ensuring no files will be uploaded to the root (or first layer as you put it), one workaround would be to set the default folder to the folder of your choice in Project Settings. Here's an example:

https://user-images.githubusercontent.com/42867097/158935225-cdf44b88-78e5-4ea1-893b-6658b2dbb943.mp4

Of course this is far from ideal as any other means of uploading that is not a File/Files/Image interface with configured folder will still upload to this folder, but hope it can somewhat help you in your current setup.

**Reply by lucameusburger** [reply]
Thanks! You are right but i think that's far from ideal. Especially because if i forget to set a folder in some file-field, everything will end up in the avatar-folder and i wouldn't even recognize.

**Comment by azrikahar** [comment]
⚠ Big disclaimer: this is purely to outline what's possible, but it is by no means a form of endorsement and should be treated as "try at your own risk" type of ordeal

---

Another pretty hacky but technically "closer" to the interaction you are looking for would be to go to the `directus` package in node_modules to patch this YAML file - `directus\dist\database\system-data\fields\users.yaml` and add the folder option using the folder UUID of your choice:

(you may be able to persist such patch using [`patch-package`](https://www.npmjs.com/package/patch-package) for example)

![image](https://user-images.githubusercontent.com/42867097/158936089-0ff9f921-903c-4a42-af70-cfdc5d295a58.png)


And it should work:

https://user-images.githubusercontent.com/42867097/158936427-f4d455e5-bd3b-4742-8166-9fc381d77db0.mp4

**Reply by benhaynes** [reply]
Nice workaround! I have had the same idea as OP for a while... I think it would be nice to have User Avatars go into a `User Avatars` sub-directory that we include out of the box. Especially since we don't (currently) allow changing that system field's options.

@azrikahar — how hard is that to add?

**Reply by lucameusburger** [reply]
Thank's a lot for this! I may implement this patch in my system since there are  a lot of new users and they all love to change their avatar every few days ;-)

**Comment by ched-dev** [comment]
Here's my use case for this feature:

I will be building a content site where all users profiles will be publicly accessible and I have to open up all files to public permissions to allow viewing of avatars. I have other folder uploads I want to keep private.

**Comment by mslonina** [comment]
Hi, 

would be possible to extend the File library field with options allowing to set folder dynamically, i.e. by content id, date - similar to Wordpress default behaviour? Then, the logic could create folder(s) automatically, if not present. It would also work for avatars (setting default folder to 'Avatars' or sth).

**Reply by rijkvanzanten** [reply]
That'd be cool! As of right now, you should be able to implement such functionality through the files.upload hook

**Reply by benhaynes** [reply]
@rijkvanzanten — could Data Flows be used for dynamic File folder routing?

1. Trigger: File Upload
2. Conditional logic to determine where it should go
3. Data Transform of File's Folder based on Step 2 Logic

**Reply by rijkvanzanten** [reply]
Yeah!

**Reply by luciandavidescu** [reply]
There's not enough info* coming with the event to determine if a file was uploaded as an avatar or not. example:

```
{
  "event": "files.upload",
  "payload": {
    "title": "Img 0926",
    "filename_download": "IMG_0926.JPG",
    "type": "image/jpeg",
    "storage": "local",
    "folder": "be53400c-4225-4954-be2f-ea0765b03ce5",
    "filename_disk": "62d9ac5d-8b29-4f21-aba3-dce7b1dc9556.JPG",
    "filesize": 135673,
    "height": 1024,
    "width": 1024,
    "metadata": {}
  },
  "key": "62d9ac5d-8b29-4f21-aba3-dce7b1dc9556",
  "collection": "directus_files"
}
```

Maybe a field indicating role/origin might help, such as the name of the field where the upload was initiated.

*Well, except for folder UUID (which would need to be hardcoded to work) or for the peculiar situation that the position of "folder" could be used as a hint (it's the first element with uploads to non-default folders, but not with default folders, where avatars are expected to be)

**Comment by rijkvanzanten** [comment]
Heya! Thanks for opening this feature request!

This feature request has received over **15 votes** from the community. This means we'll move this feature request to the Under Review state! 

The Core team will schedule a meeting to review this request as soon as possible. The discussion will then be approved or denied. You may or may not be invited to join this meeting with the core team.

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._

**Comment by NilsBaumgartner1994** [comment]
I would suggest adding the option to automatically generate a folder if it is missing.
